### PR TITLE
Add a type for ensure on service resources

### PR DESCRIPTION
**Pull request by npwalker**

### CODE REVIEW

**Comment by ekohl** [comment]
Would ServiceEnsure or EnsureService be easier to read?

**Comment by jonhattan** [comment]
Only first character uppercase, for consistency with existing types (f.e: `Stdlib::Absolutepath`).

`Ensureservice` seems better to me, so it's consistent with adding `Ensurewhatever` in a future.

**Comment by ekohl** [comment]
A third option is ```Stdlib::Ensure::Service``` and ```Stdlib::Ensure::Whatever```.

File: types/serviceensure.pp [hunk 1]
```diff
@@ -0,0 +1 @@
+type Stdlib::Serviceensure = Enum['stopped', 'running']
```
Comment (b4ldr) [inline types/serviceensure.pp]: should this not be 
```ruby
type Stdlib::Serviceensure = Varient[Boolean,Enum['stopped', 'running']]
```

Comment (npwalker) [reply types/serviceensure.pp]: IMO, using `ensure => true` or `ensure => false` on a service is an anti-pattern.  It works but it's just not as clear to new users what is going on.  As a convention I don't support it.

Comment (b4ldr) [reply types/serviceensure.pp]: I tend to agree with you however if this is going into stdlib then i think it should conform to the documentation 
https://puppet.com/docs/puppet/5.3/type.html#service-attribute-ensure

**Comment by david22swan** [comment]
Add the needed docs change and I will merge.

**Comment by david22swan** [comment]
@npwalker Moving towards a clean release on this module to allow a full rubocop run, so would love to get this in before that. Please add the docs so I can do o, otherwise a full-rebase will be needed.

**Comment by npwalker** [comment]
@david22swan okay updated.

**Pull request merged**